consumer/consumererror/signalerrors.go

```diff
@@ -0,0 +1,123 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consumererror
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+// Traces is an error that may carry associated Trace data for a subset of received data
+// that faiiled to be processed or sent.
+type Traces struct {
+	error
+	failed pdata.Traces
+}
+
+// NewTraces creates a Traces that can encapsulate received data that failed to be processed or sent.
+func NewTraces(err error, failed pdata.Traces) error {
+	return Traces{
+		error:  err,
+		failed: failed,
+	}
+}
+
+// IsTraces checks if an error includes a Traces.
+func IsTraces(err error) bool {
```

Comment:
In case of this I think user need access to the instance, so probably better to make this `AsTraces` and follow the signature of `errors.As`

Reply:
Done.